platform/sleep.h

```diff
@@ -0,0 +1,93 @@
+
+/** \addtogroup platform */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_SLEEP_H
+#define MBED_SLEEP_H
+
+#include "sleep_api.h"
+
+#if !DEVICE_SLEEP
+#warning Sleep is not supported on this platform.
+#else /* DEVICE_SLEEP */
+#ifndef NDEBUG
+#warning Sleep is disabled for debug builds.
```

Comment:
It's not like our builds are clean of warnings. And yes default build is debug, which is probably the correct behavior for mbed-os. So you think that the documentation note (which i will submit separately to the handbook) is enough of a info?